DataDog/datadog-lambda-go
# Do not restart tracer on every invocation

### What does this PR do?

Update dd-trace-go to `v1.30.0`. Use the new `Flush()` method to flush traces after every invocation without having to stop and restart the tracer each time.

### Motivation

This will improve the performance of the library on warm starts because the tracer will not have to be restarted. It will have no effect on cold start performance because the tracer does need to start the first time.

### Testing Guidelines

I tested the performance change by logging timestamps before and after the tracer initialized and noting the performance improvements from using the `Flush()` method. The integration tests confirm that the traces are appropriately logged.

### Types of changes


### Checklist
